### PR TITLE
Bulk update #127

### LINKED ISSUE #127: Bulk Update
Currently only bulk create is supported by the package, where an IEnumerable<TItem> can be passed to CreateAsync function and items are added to the cosmos container. This functionality is missing for UpdateAsync. It comes in handy when you have multiple items that needs to be added and increases performance.

Add a new UpdateAsync function that accepts IEnumerable<TItem> and updates the cosmos container values
UpdateAsync(IEnumerable<TItem> values, CancellationToken cancellationToken = default);

I cloned the repo and added the function locally on my machine, tested and working as expected. 

If this is a feature that you see valuable, I can create a PR for review.

### PR DESCRIPTION
**Pull request by mustafarabie**
Added logic to enable bulk update to cosmos repo

Closes #127

### CODE REVIEW

File: Microsoft.Azure.CosmosRepository/src/DefaultRepository.cs [hunk 1]
```diff
@@ -168,6 +168,20 @@ await container.UpsertItemAsync<TItem>(value, new PartitionKey(value.PartitionKe
...
+        /// <inheritdoc/>
+        public async ValueTask<IEnumerable<TItem>> UpdateAsync(
+            IEnumerable<TItem> values,
+            CancellationToken cancellationToken = default)
+        {
+            IEnumerable<Task<TItem>> updateTasks =
+                values.Select(value => UpdateAsync(value, cancellationToken).AsTask())
+                    .ToList();
+
+            _ = await Task.WhenAll(updateTasks).ConfigureAwait(false);
```
Comment (mumby0168) [inline Microsoft.Azure.CosmosRepository/src/DefaultRepository.cs:180]: I think we can drop the `_ = ` assignment.

```suggestion
            await Task.WhenAll(updateTasks).ConfigureAwait(false);

```

File: Microsoft.Azure.CosmosRepository/src/InMemoryRepository.cs [hunk 2]
```diff
@@ -105,6 +105,19 @@ public async ValueTask<TItem> UpdateAsync(TItem value, CancellationToken cancell
             return Items[value.Id] = value;
         }
 
+        /// <inheritdoc/>
+        public async ValueTask<IEnumerable<TItem>> UpdateAsync(IEnumerable<TItem> values, CancellationToken cancellationToken = default)
```
Comment (mumby0168) [inline Microsoft.Azure.CosmosRepository/src/InMemoryRepository.cs:109]: Looks good, there are some tests for this class that would need updating if you don't mind.

**Review by mumby0168 — CHANGES_REQUESTED** [review]
Looks good just a few little things, but looks good thanks for the PR 😃

**Comment by mumby0168** [comment]
Hey @all-contributors please add  @mustafarabie for code

File: Microsoft.Azure.CosmosRepository/src/InMemoryRepository.cs (hunk 2 above)
Comment (mustafarabie) [reply Microsoft.Azure.CosmosRepository/src/InMemoryRepository.cs:109]: Thanks for the quick review. Will add tests and create a PR.

Comment (mustafarabie) [reply Microsoft.Azure.CosmosRepository/src/InMemoryRepository.cs:109]: Updated PR with tests.

**Review by mumby0168 — APPROVED** [review]
Looks good to me 👍

**Comment by mumby0168** [comment]
Hey @all-contributors please add @mustafarabie for tests

File: Microsoft.Azure.CosmosRepository/src/DefaultRepository.cs [hunk 3]
```diff
@@ -168,6 +168,21 @@ await container.UpsertItemAsync<TItem>(value, new PartitionKey(value.PartitionKe
...
+        public async ValueTask<IEnumerable<TItem>> UpdateAsync(
+            IEnumerable<TItem> values,
+            CancellationToken cancellationToken = default)
+        {
+            IEnumerable<Task<TItem>> updateTasks =
+                values.Select(value => UpdateAsync(value, cancellationToken).AsTask())
+                    .ToList();
+
+            await Task.WhenAll(updateTasks).ConfigureAwait(false);
+
```
Comment (IEvangelist) [inline Microsoft.Azure.CosmosRepository/src/DefaultRepository.cs:181]:
```suggestion
```

**Review by IEvangelist — DISMISSED** [review]

**Comment by mumby0168** [comment]
Thanks for the great work @mustafarabie checkout the release your changes will be available in v2.6.0 🎉 

https://github.com/IEvangelist/azure-cosmos-dotnet-repository/releases/tag/2.6.0

**Pull request merged**